webapi/src/main/scala/org/knora/webapi/messages/store/triplestoremessages/TriplestoreMessages.scala

```diff
@@ -235,39 +232,44 @@ case class BlankNodeLiteralV2(value: String) extends LiteralV2 {
   override def toString: String = value
 }
 
-/**
- * Represents a string with language iso. Allows sorting inside collections by the value.
- *
- * @param value    the string value.
- * @param language the language iso.
- */
-case class StringLiteralV2 private (value: String, language: Option[String])
-    extends LiteralV2
-    with OntologyLiteralV2
-    with Ordered[StringLiteralV2] {
-
+sealed trait StringLiteralV2 extends LiteralV2 with OntologyLiteralV2 with Ordered[StringLiteralV2] {
+  def value: String
   override def compare(that: StringLiteralV2): Int = this.value.compareTo(that.value)
   override def toString: String                    = value
-
-  def languageCode: Option[LanguageCode] = language.map(LanguageCode.from).flatMap(_.toOption)
-
-  def toRdfLiteral: StringLiteral =
-    language.map(Rdf.literalOfLanguage(value, _)).getOrElse(Rdf.literalOf(value))
+  def languageOption: Option[LanguageCode] = this match {
+    case LanguageTaggedStringLiteralV2(_, language) => Some(language)
+    case PlainStringLiteralV2(_)                    => None
+  }
 }
 
+case class LanguageTaggedStringLiteralV2(value: String, language: LanguageCode) extends StringLiteralV2
+case class PlainStringLiteralV2(value: String)                                  extends StringLiteralV2
+
 object StringLiteralV2 {
   implicit val codec: JsonCodec[StringLiteralV2] = DeriveJsonCodec.gen[StringLiteralV2]
   implicit val schema: Schema[StringLiteralV2]   = Schema.derived[StringLiteralV2]
 
-  def from(value: String, lang: LanguageCode): StringLiteralV2 =
-    from(value, Some(lang.value))
-
-  def from(value: String, language: Option[String]): StringLiteralV2 = language match {
-    case Some(_) if value.isEmpty => throw BadRequestException("String value is missing.")
-    case _                        => StringLiteralV2(value, language)
+  val orderByValue: Ordering[StringLiteralV2] = Ordering.by(_.value)
+  val orderByLanguage: Ordering[StringLiteralV2] = Ordering.by {
+    case LanguageTaggedStringLiteralV2(_, language) => language.value
+    case PlainStringLiteralV2(_)                    => ""
   }
```

Comment:
Comment: Not for this PR, but this seems like a non-meaningful operation on `PlainStringLiteral`.